subsys/net/lib/coap/coap_server.c

```diff
@@ -127,7 +127,8 @@ static int coap_service_remove_observer(const struct coap_service *service,
 
 static int coap_server_process(int sock_fd)
 {
-	uint8_t buf[CONFIG_COAP_SERVER_MESSAGE_SIZE];
+	static uint8_t buf[CONFIG_COAP_SERVER_MESSAGE_SIZE];
```

Comment:
Wouldn't there be now problem if we receive data from multiple clients?

Reply:
There's a single thread running this function based on `poll` events from multiple receiving sockets.